dashboard/src/components/ModelField.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <div class="model-field">
+    <b>{{ title }}:</b>
+    <input v-if="inputType == 'number'" type="number" v-model.number="$parent.parameters[fieldName]" />
+    <input v-if="inputType != 'number'" v-model="$parent.parameters[fieldName]" />
+    <help :helpText="$parent.definitions.properties[fieldName].description" />
+    <br />
+    <span style="color: #F00;" v-if="fieldName in $parent.errors">
+      {{ $parent.errors[fieldName] }}
+    </span>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+
+@Component
+export default class ModelField extends Vue{
+  @Prop() title!: string;
+  @Prop() inputType?: string;
+  @Prop() fieldName!: string;
+  //@Prop() parameters!: Record<string, any>;
+  //@Prop() errors!: Record<string, any>;
+  //@Prop() definitions!: Record<string, any>;
+
+}
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style scoped>
+
+</style>
```

Comment:
@alorenzo175 I created this component to factor out the copy pasta. The way it's written here, it reads and alters the props of its parents directly. We could also use the commented out props and pass all of the props in. I think this should be okay since this will always be a leaf node with no children, and I can't think of any issues it may cause. Do you have any objections or concerns with this?

Reply:
I'm definitely just learning Vue, so how is this altering the parent?

Also, I don't think the errors span is working. For example, modifying models.py adding `ge=0` to `u_c` and then entering -1 for uC does not generate 'should be >= 0'

Reply:
Ah, thanks. there was no validate function for TemperatureParameters. errors should work now, an empty field will yield "should be a number".

The `v-model` bindings are accessing the `$parent` directly. Essentially, anywhere this is used is agreeing to have it's `parameters` mutated by this component. It's not too far from what the rest of these components are doing, e.g. clicking "add inverter" appends a new inverter object to the systems `inverters` property, but hands it off to a component to be altered.

To explain why I'm double checking, All of the model components break from the suggested Vue pattern of passing props down through the hierarchy, and emitting events up to mutate data. But since these components are meant to mutate the system specification, which will only exist in memory on the page, it seems like a logical way to go.

Reply:
I see. I guess that's OK. good to go change the rest to use this component